src/main/java/org/mule/service/http/impl/service/util/RedirectUtils.java

```diff
@@ -51,35 +59,54 @@ public static boolean isRedirected(int statusCode) {
   }
 
   /**
-   * @param requestMethod
    * @param responseStatusCode
-   * @return the original request method if the status code is 301, 307 or 308, or GET if it is 302 or 303.
+   * @return false if the status code is 301, 307 or 308, or true if it is 302 or 303.
    */
-  public static String getMethodForStatusCode(String requestMethod, int responseStatusCode) {
+  public static boolean sendAsGet(int responseStatusCode) {
     switch (responseStatusCode) {
       case 301:
       case 307:
       case 308:
-        return requestMethod;
+        return false;
       case 302:
       case 303:
-        return HttpConstants.Method.GET.name();
+        return true;
       default:
         throw new IllegalArgumentException("Invalid status code");
     }
   }
 
   /**
    * Create a new request with the params of the original and the new URI from the LOCATION header.
-   * 
+   *
    * @param response HttpResponse
    * @param request  HttpRequest
    * @return an HttpRequest request.
    */
-  public static HttpRequest createRedirectRequest(HttpResponse response, HttpRequest request) {
+  public static HttpRequest createRedirectRequest(HttpResponse response, HttpRequest request, HttpRequestOptions options) {
```

Comment:
Add a comment about why you don't handle cookies here, and link to the method that handles them